packages/vite-plugin/src/proxy-middleware.ts

```diff
@@ -0,0 +1,244 @@
+/**
+ * This modules provides the proxy located at
+ * /motion-canvas-proxy/...
+ *
+ * It is needed when acessing remote resources.
+ * Trying to access remote resources works while
+ * in preview, but will fail when you try to
+ * output the image (= "read" the canvas)
+ *
+ * See https://developer.mozilla.org/en-US/docs/Web/HTML/CORS_enabled_image
+ * for reasons
+ *
+ * Using the proxy circumvents CORS-issues because
+ * this way all remote resources are served from the
+ * same host as the main app.
+ */
+
+import {Connect} from 'vite';
+import axios from 'axios';
```

Comment:
`axios` is a dependency of the `docs` package. It works for you locally, because in a monorepo dependencies are shared in root node_modules. After publishing, this will fail because npm won't install `axios`.

`axios` is a quite heavy dependency and I'm not sure if we should add it to `vite-plugin`. Can't we just use Node's `https` package?

Reply:
We can! My thought behind using axios was that I would not need to handle all the edge cases, like HTTP Redirects.
I should be able to get rid of the dependency though

Reply:
@aarthificial 

How would you feel about depending on `follow-redirects` ?

https://github.com/follow-redirects/follow-redirects

It is [28.3 kB unpacked](https://www.npmjs.com/package/follow-redirects) according to npm  while [axios is 1.72MB](https://www.npmjs.com/package/axios)

Reply:
I'm okay with that